extension/parquet/parquet_writer.cpp

```diff
@@ -578,7 +578,50 @@ void ParquetWriter::Flush(ColumnDataCollection &buffer) {
 	}
 
 	PreparedRowGroup prepared_row_group;
-	PrepareRowGroup(buffer, prepared_row_group);
+
+	bool needs_transform = false;
+	D_ASSERT(buffer.ColumnCount() == column_writers.size());
+	for (idx_t col_idx = 0; col_idx < column_writers.size(); col_idx++) {
+		auto &column_writer = *column_writers[col_idx];
+		if (!column_writer.HasTransform()) {
+			continue;
+		}
+		needs_transform = true;
+	}
+
+	if (needs_transform) {
+		vector<LogicalType> transformed_buffer_types;
+		vector<unique_ptr<Expression>> transform_expressions;
+		D_ASSERT(buffer.ColumnCount() == column_writers.size());
+		for (idx_t col_idx = 0; col_idx < column_writers.size(); col_idx++) {
+			auto &column_writer = *column_writers[col_idx];
+			auto &original_type = buffer.Types()[col_idx];
+			auto expr = make_uniq<BoundReferenceExpression>(original_type, col_idx);
+			if (!column_writer.HasTransform()) {
+				transformed_buffer_types.push_back(original_type);
+				transform_expressions.push_back(std::move(expr));
+				continue;
+			}
+			transformed_buffer_types.push_back(column_writer.TransformedType());
+			transform_expressions.push_back(column_writer.TransformExpression(std::move(expr)));
+		}
+
+		ColumnDataCollection transformed_buffer(GetContext(), transformed_buffer_types,
```

Comment:
This is currently created once per Flush, it should probably be a `unique_ptr<ColumnDataCollection> transform_buffer` that's created once, somewhere in an initialize method